vision/google/cloud/vision/batch.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Batch multiple images into one request."""
+
+
+class Batch(object):
```

Comment:
@lukesneeringer @daspecster You can add a state to the batch and "tombstone" it once committed. Then the return value can be accessed as an attribute of the batch. The state makes it so that we don't accidentally over-ride the stored return value.

(We tombstone datastore transactions / maybe batches too in datastore.)